### ISSUE TITLE
Some commands that handle ranges don't account for the step value

### ISSUE THREAD

**Issue by TWSiO** [nushell/nushell#11521]
### Describe the bug

When you use a range with a step/second value (the middle parameter when using 3 parameters), some of the commands such as `range` and `str substring` ignore the step value.

```nushell
> [5 6 7 8] | range 0..2..3
╭───┬───╮
│ 0 │ 5 │
│ 1 │ 6 │
│ 2 │ 7 │
│ 3 │ 8 │
╰───┴───╯
```

```nushell
> 'abcdef' | str substring 0..2..5
abcde
```

Other commands that use ranges might potentially have the same issue, I just haven't gone through them all.

### How to reproduce

See above

### Expected behavior

```nushell
> [5 6 7 8] | range 0..2..3
╭───┬───╮
│ 0 │ 5 │
│ 1 │ 7 │
╰───┴───╯
```

```nushell
'abcdef' | str substring 0..2..5
ace
```

### Configuration

| key                | value                                                |
| ------------------ | ---------------------------------------------------- |
| version            | 0.88.1                                               |
| branch             |                                                      |
| commit_hash        |                                                      |
| build_os           | linux-x86_64                                         |
| build_target       | x86_64-unknown-linux-gnu                             |
| rust_version       | rustc 1.75.0 (82e1608df 2023-12-21) (Homebrew)       |
| cargo_version      | cargo 1.75.0                                         |
| build_time         | 2023-12-14 17:14:47 +00:00                           |
| build_rust_channel | release                                              |
| allocator          | mimalloc                                             |
| features           | dataframe, default, extra, sqlite, trash, which, zip |
| installed_plugins  | from eml, from ics, from ini, from vcf               |

### Additional context

A previous issue comment noted an idea for how to handle the underlying inconsistancies: https://github.com/nushell/nushell/issues/11271#issuecomment-1848284555

**Comment by IanManske** [comment]
Yep, almost no commands handle the case when a range has a step value. The step value is notoriously inconsistent and confusing. Issue #11086 is related where it was mentioned that:
```nushell
> 1..100 == 1..2..100
true
```

IMO, the range step functionality isn't worth keeping when the `seq` and `every` commands can accomplish the same thing.
